src/routes/clients.js

```diff
@@ -61,14 +71,26 @@ router.post('/', (req, res) => {
       case 'name':
         errno = ERRNO_INVALID_API_CLIENT_NAME;
         break;
+      case 'permissions':
+        errno = ERRNO_INVALID_API_CLIENT_PERMISSION;
+        break;
       default:
         errno = ERRNO_BAD_REQUEST;
     }
     return ApiError(res, 400, errno, BAD_REQUEST);
   }
 
   db().then(models => {
-    models.Clients.create(req.body).then(client => {
+    let client;
+    models.sequelize.transaction(transaction => {
+      return models.Clients.create(req.body, { transaction }).then(_client => {
+        client = _client;
+        if (!req.body.permissions) {
+          return;
+        }
+        return client.addPermissions(req.body.permissions, { transaction });
+      });
```

Comment:
This was my first approach, but somehow I failed to apply it... so if you don't mind, I'll stick with the current approach.